vita3k/renderer/src/gl/renderer.cpp

```diff
@@ -482,19 +482,29 @@ void get_surface_data(GLContext &context, size_t width, size_t height, size_t st
     }
 
     glPixelStorei(GL_PACK_ROW_LENGTH, static_cast<GLint>(stride_in_pixels));
-    if (format == SCE_GXM_COLOR_FORMAT_U8U8U8U8_ABGR) {
+    switch (format) {
+    case SCE_GXM_COLOR_FORMAT_U8U8U8U8_ABGR:
+    case SCE_GXM_COLOR_FORMAT_U8U8U8U8_ARGB:
         glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RGBA, GL_UNSIGNED_BYTE, pixels);
-    } else if (format == SCE_GXM_COLOR_FORMAT_U8U8U8U8_RGBA) {
+        break;
+    case SCE_GXM_COLOR_FORMAT_U8U8U8U8_RGBA:
         glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RGBA, GL_UNSIGNED_BYTE, pixels);
-        for (int i = 0; i < width * height; ++i) {
+        for (int i = 0; i < width * height; i++) {
             uint8_t *pixel = reinterpret_cast<uint8_t *>(&pixels[i]);
             std::swap(pixel[0], pixel[3]);
             std::swap(pixel[1], pixel[2]);
         }
-    } else {
-        // TODO wise color conversino implementation
-        // maybe we can use PBO?
-        assert(false);
+        break;
+    case SCE_GXM_COLOR_FORMAT_U8U8U8_BGR:
+        glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RGB, GL_UNSIGNED_BYTE, pixels);
+        break;
+    case SCE_GXM_COLOR_FORMAT_U8_R:
+        glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RED, GL_UNSIGNED_BYTE, pixels);
+        break;
+    default:
+        LOG_ERROR("Color format not implemented: {}, report this to developer", format);
```

Comment:
We don't need reports for this. We can just complete every color format if we do it wisely.

Reply:
It's maybe that they ignore a part when its using three components. If the stride_in_pixels is the same as in 4 components, this is the right way. We can just swap colors here assuming color format were rendering is bgr. (Thats why using just gl_rgb was working, as it doesnt need to be swapped and it should read first 3 pixel in every stride)

Reply:
No, i no like add all without any test, so i add it obe by one if game request
For can see what works in réal time

Reply:
Im going to do it

Reply:
my case, and just with see here for T@g and Home Arcade, need test case by case for all new color ,format request, for don't write wrong code color

Reply:
Adding each code that works for specific game is workaround. We need to really understand what's going on.

Reply:
i see, but for no, for both, only works like that ^^